command/v6/create_service_broker_command_test.go

```diff
@@ -0,0 +1,204 @@
+package v6_test
+
+import (
+	"errors"
+
+	"code.cloudfoundry.org/cli/actor/actionerror"
+	"code.cloudfoundry.org/cli/actor/v2action"
+	"code.cloudfoundry.org/cli/command/commandfakes"
+	"code.cloudfoundry.org/cli/command/flag"
+	"code.cloudfoundry.org/cli/command/translatableerror"
+	. "code.cloudfoundry.org/cli/command/v6"
+	"code.cloudfoundry.org/cli/command/v6/v6fakes"
+	"code.cloudfoundry.org/cli/util/configv3"
+	"code.cloudfoundry.org/cli/util/ui"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+)
+
+var _ = Describe("create-service-broker Command", func() {
+	var (
+		cmd             CreateServiceBrokerCommand
+		testUI          *ui.UI
+		fakeConfig      *commandfakes.FakeConfig
+		fakeSharedActor *commandfakes.FakeSharedActor
+		fakeActor       *v6fakes.FakeCreateServiceBrokerActor
+		binaryName      string
+		executeErr      error
+		spaceScoped     bool
+		extraArgs       []string
+	)
+
+	BeforeEach(func() {
+		testUI = ui.NewTestUI(nil, NewBuffer(), NewBuffer())
+		fakeConfig = new(commandfakes.FakeConfig)
+		fakeSharedActor = new(commandfakes.FakeSharedActor)
+		fakeActor = new(v6fakes.FakeCreateServiceBrokerActor)
+
+		binaryName = "faceman"
+		fakeConfig.BinaryNameReturns("faceman")
+		spaceScoped = false
+		extraArgs = nil
+		cmd = CreateServiceBrokerCommand{
+			UI:          testUI,
+			Config:      fakeConfig,
+			SharedActor: fakeSharedActor,
+			Actor:       fakeActor,
+			RequiredArgs: flag.ServiceBrokerArgs{
+				ServiceBroker: "cool-broker",
+				Username:      "admin",
+				Password:      "password",
+				URL:           "https://broker.com",
+			},
+			SpaceScoped: spaceScoped,
+		}
+
+	})
+
+	JustBeforeEach(func() {
+		executeErr = cmd.Execute(extraArgs)
+	})
+
+	When("the user provides arguments", func() {
+		BeforeEach(func() {
+			extraArgs = []string{"some-extra-arg"}
+		})
+
+		It("fails with a TooManyArgumentsError", func() {
+			Expect(executeErr).To(MatchError(translatableerror.TooManyArgumentsError{
+				ExtraArgument: "some-extra-arg",
+			}))
+		})
+	})
+
+	When("the user does not have an org and space targeted", func() {
+		BeforeEach(func() {
+			fakeSharedActor.CheckTargetReturns(nil)
+			fakeConfig.CurrentUserReturns(
+				configv3.User{Name: "some-user"},
+				nil)
+		})
+
+		When("fetching the current user fails", func() {
+			BeforeEach(func() {
+				fakeConfig.CurrentUserReturns(configv3.User{}, errors.New("no user"))
+			})
+
+			It("returns an error", func() {
+				Expect(executeErr).To(MatchError("no user"))
+			})
+		})
+
+		Context("not space scoped", func() {
+			When("creating a service broker is successful", func() {
+				BeforeEach(func() {
+					fakeActor.CreateServiceBrokerReturns(v2action.ServiceBroker{}, []string{"a-warning", "another-warning"}, nil)
+				})
+
+				It("no error is returned from the actor", func() {
```

Comment:
This It description is describing what the Actor does rather then the object under test; probably ought to change this so that it's describing what the Command does.